app/packages/viz3d/src/point_cloud_looker.ts

```diff
@@ -0,0 +1,346 @@
+        return new Promise((resolve,_) => {
+            resolve(mesh);
+        })
+    }
+    else {
+        return _global_draco_loader.then((loader) => {
+            return loader.loadRemoteMesh(path);
+        }).then((mesh) => {
+            _global_mesh_cache.set(path, mesh);
+            return mesh;
+        });
+    }
+}
+
+function create3DDisplay (config: pcd.SceneConfig): pcd.Display3D<HTMLCanvasElement> {
+    // TODO: Gross hack
+    if (!_global_3D_display.created()){
+        // https://github.com/mrdoob/three.js/blob/master/examples/webgl_multiple_canvases_circle.html
+        // All modal display happens on this canvas, and is copied over to the looker elements' canvases
+        const canvas = document.createElement("canvas");
+        return _global_3D_display.make(canvas, config);
+    }
+    else return _global_3D_display.make(null, config);
+}
+
+export class PointCloudConfig implements BaseConfig {
+    thumbnail: boolean = true;
+    thumbnail_width: number = 100;
+    thumbnail_height: number = 100;
+    src: string = "";
+    dimensions: Dimensions = [0, 0];
+    sampleId: string = "";
+    fieldSchema: Schema;
+    displayConfig: pcd.SceneConfig;
+
+    constructor() {
+        this.displayConfig = DEFAULT_3D_DISPLAY_CONFIG;
+    }
+};
+
+export interface PointCloudState extends BaseState {
+    config: PointCloudConfig;
+};
+
+export const DEFAULT_POINTCLOUD_STATE: PointCloudState = {
+    config: new PointCloudConfig,
+    disabled: false,
+    cursorCoordinates: [0, 0],
+    pixelCoordinates: [0, 0],
+    disableControls: false,
+    loaded: false,
+    hovering: false,
+    hoveringControls: false,
+    showControls: false,
+    showOptions: false,
+    options: undefined,
+    scale: 0,
+    pan: [0, 0],
+    panning: false,
+    rotate: 0,
+    strokeWidth: 0,
+    fontSize: 0,
+    wheeling: false,
+    windowBBox: [0, 0, 0, 0],
+    transformedWindowBBox: [0, 0, 0, 0],
+    mediaBBox: [0, 0, 0, 0],
+    transformedMediaBBox: [0, 0, 0, 0],
+    canvasBBox: [0, 0, 0, 0],
+    textPad: 0,
+    pointRadius: 0,
+    dashLength: 0,
+    relativeCoordinates: [0, 0],
+    mouseIsOnOverlay: false,
+    showHelp: false,
+    overlaysPrepared: false,
+    disableOverlays: false,
+    zoomToContent: false,
+    setZoom: false,
+    hasDefaultZoom: false,
+    SHORTCUTS: undefined,
+    error: 0,
+    destroyed: false,
+    reloading: false
+};
+
+export class PointCloudSample implements Sample {
+    metadata: { width: number; height: number; };
+    id: string = "";
+    media_type: "model" = "model";
+    filepath: string = "";
+    compressed_path: string = "";
+    tags: string[] = [];
+    _label_tags: string[] = [];
+
+};
+
+export interface PointCloudOptions extends BaseOptions {
+    zoom: boolean;
+};
+
+export const DEFAULT_POINTCLOUD_OPTIONS = {
+    ...DEFAULT_BASE_OPTIONS,
+    zoom: false
+}
+
+export class PointCloudElement extends BaseElement<PointCloudState, HTMLCanvasElement>{
```

Comment:
Rename this and `PointCloudLooker` to something more general. This can handle all 3D data.